sdk/src/main/java/com/hedera/hashgraph/sdk/TokenRejectTransaction.java

```diff
@@ -0,0 +1,237 @@
+     * @return the list of tokenIds.
+     */
+    public List<TokenId> getTokenIds() {
+        return tokenIds;
+    }
+
+    /**
+     * Assign the list of tokenIds.
+     * @param tokenIds the list of tokenIds.
+     * @return {@code this}
+     */
+    public TokenRejectTransaction setTokenIds(List<TokenId> tokenIds) {
+        requireNotFrozen();
+        Objects.requireNonNull(tokenIds);
+        this.tokenIds = new ArrayList<>(tokenIds);
+        return this;
+    }
+
+    /**
+     * Add a token to the list of tokens.
+     * @param tokenId token to add.
+     * @return {@code this}
+     */
+    public TokenRejectTransaction addTokenId(TokenId tokenId) {
+        requireNotFrozen();
+        tokenIds.add(tokenId);
+        return this;
+    }
+
+    /**
+     * Extract the list of nftIds.
+     * @return the list of nftIds.
+     */
+    public List<NftId> getNftIds() {
+        return nftIds;
+    }
+
+    /**
+     * Assign the list of nftIds.
+     * @param nftIds the list of nftIds.
+     * @return {@code this}
+     */
+    public TokenRejectTransaction setNftIds(List<NftId> nftIds) {
+        requireNotFrozen();
+        Objects.requireNonNull(nftIds);
+        this.nftIds = new ArrayList<>(nftIds);
+        return this;
+    }
+
+    /**
+     * Add a nft to the list of nfts.
+     * @param nftId nft to add.
+     * @return {@code this}
+     */
+    public TokenRejectTransaction addNftId(NftId nftId) {
+        requireNotFrozen();
+        nftIds.add(nftId);
+        return this;
+    }
+
+    /**
+     * Build the transaction body.
+     *
+     * @return {@link com.hedera.hashgraph.sdk.proto.TokenRejectTransactionBody}
+     */
+    TokenRejectTransactionBody.Builder build() {
+        var builder = TokenRejectTransactionBody.newBuilder();
+
+        if (ownerId != null) {
+            builder.setOwner(ownerId.toProtobuf());
+        }
+
+        for (TokenId tokenId : tokenIds) {
+            builder.addRejections(TokenReference.newBuilder().setFungibleToken(tokenId.toProtobuf()).build());
+        }
+
+        for (NftId nftId : nftIds) {
+            builder.addRejections(TokenReference.newBuilder().setNft(nftId.toProtobuf()).build());
+        }
+
+        return builder;
+    }
+
+    /**
+     * Initialize from the transaction body.
+     */
+    void initFromTransactionBody() {
+        var body = sourceTransactionBody.getTokenReject();
+        if (body.hasOwner()) {
+            ownerId = AccountId.fromProtobuf(body.getOwner());
+        }
+
+        for (TokenReference tokenReference : body.getRejectionsList()) {
+            if (tokenReference.hasFungibleToken()) {
+                tokenIds.add(TokenId.fromProtobuf(tokenReference.getFungibleToken()));
+            } else if (tokenReference.hasNft()) {
+                nftIds.add(NftId.fromProtobuf(tokenReference.getNft()));
+            }
+        }
+    }
+
+    @Override
+    void validateChecksums(Client client) throws BadEntityIdException {
```

Comment:
Should we add here checksum for NFTs as well? Something like this `nftID.getTokenId().validateChecksum(client)`

Reply:
Added, thanks for noticing